messaging/src/test/java/org/axonframework/eventhandling/MultiSourceTrackingTokenTest.java

```diff
@@ -0,0 +1,156 @@
+package org.axonframework.eventhandling;
+
+import org.junit.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class MultiSourceTrackingTokenTest {
+
+    MultiSourceTrackingToken testSubject;
+
+    @Before
+    public void setUp() {
+        Map<String, TrackingToken> tokenMap = new HashMap<>();
+        tokenMap.put("token1", new GlobalSequenceTrackingToken(0));
+        tokenMap.put("token2", new GlobalSequenceTrackingToken(0));
+
+        testSubject = new MultiSourceTrackingToken(tokenMap);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIncompatibleToken() {
+        testSubject.covers(new GlobalSequenceTrackingToken(0));
+    }
+
+    @Test
+    public void lowerBound() {
+        Map<String, TrackingToken> newTokens = new HashMap<>();
+        newTokens.put("token1", new GlobalSequenceTrackingToken(1));
+        newTokens.put("token2", new GlobalSequenceTrackingToken(2));
+
+        Map<String, TrackingToken> expectedTokens = new HashMap<>();
+        expectedTokens.put("token1", new GlobalSequenceTrackingToken(0));
+        expectedTokens.put("token2", new GlobalSequenceTrackingToken(0));
+        ;
+
+        MultiSourceTrackingToken newMultiToken = (MultiSourceTrackingToken) testSubject
+                .lowerBound(new MultiSourceTrackingToken(newTokens));
+
+        assertEquals(new MultiSourceTrackingToken(expectedTokens), newMultiToken);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void lowerBoundMismatchTokens() {
+        Map<String, TrackingToken> newTokens = new HashMap<>();
+        newTokens.put("token1", new GlobalSequenceTrackingToken(1));
+        newTokens.put("token3", new GlobalSequenceTrackingToken(2));
+
+        Map<String, TrackingToken> expectedTokens = new HashMap<>();
+        expectedTokens.put("token1", new GlobalSequenceTrackingToken(0));
+        expectedTokens.put("token2", new GlobalSequenceTrackingToken(0));
+        ;
+
+        MultiSourceTrackingToken newMultiToken = (MultiSourceTrackingToken) testSubject
+                .lowerBound(new MultiSourceTrackingToken(newTokens));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void lowerBoundDifferentNumberOfTokens() {
```

Comment:
I think the `upperBound()` and `covers()` method would also benefit from a different number of tokens test.